packages/babel-plugin-block/macro.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+const { createMacro } = require( 'babel-plugin-macros' );
+const { existsSync, readFileSync } = require( 'fs' );
+const { mapKeys, pick } = require( 'lodash' );
+const { dirname, join, relative } = require( 'path' );
+
+const i18nVariable = '_x';
+const aliases = {
+	styleVariations: 'styles',
+};
+const whitelistedProperties = [
+	'name',
+	'title',
+	'category',
+	'parent',
+	'icon',
+	'description',
+	'keywords',
+	'attributes',
+	'styles',
+];
+const translatableProperties = [
+	'title',
+	'description',
+	'keywords',
+	'styles',
+];
+
+function getFilename( [ filenamePath ], state ) {
+	const filename = filenamePath.evaluate().value;
+
+	return join(
+		relative( process.cwd(), dirname( state.file.opts.filename ) ),
+		filename
+	);
+}
+
+function readMetadata( filename ) {
+	if ( ! existsSync( filename ) ) {
+		throw new Error( `Invalid file name provided: ${ filename }.` );
+	}
+
+	const metadataRaw = readFileSync( filename, 'utf8' );
```

Comment:
Is there any way to make this non-blocking?

Reply:
I looked at 2 or 3 plugins that read JSON files and they use a blocking approach. I'm happy to use the async version but I would need some guidelines or any working example. Do you know anything that would be useful?